src/DalService.ts

```diff
@@ -0,0 +1,26 @@
+import { AxiosResponse } from 'axios';
+import GenerateApiError from './utils/GenerateApiError';
+
+export default class DalService {
```

Comment:
Can you rename this to like ChannelApeAPI Service? DAL service is kind of a weird name that isn't really that clear. Once you do that, i'm good with these changes.